Code refactoring

LunyaticCat opened the pull request:
## Short roundup of the initial problem
All files are currently in the root directory, making it really hard for new devs to join the project.

## What will change with this Pull Request?
every cockatrice files are put into different folders matching what they do.

BruebachL commented on cockatrice/src/card_database_parser/cockatrice_xml3.cpp:
Maybe add another underscore for readability (e.g. cockatrice_xml_3.cpp)

BruebachL commented on cockatrice/src/card_database_parser/cockatrice_xml4.cpp:
Maybe add another underscore for readability (e.g. cockatrice_xml_3.cpp)

BruebachL commented on cockatrice/src/dialogs/dlg_connect.cpp:
Some unwanted changes to UI naming here.

BruebachL reviewed (changes requested):
LGTM, pending changes.

tooomm commented:
Where does the 10k reduced lines come from?

LunyaticCat commented:
> Where does the 10k reduced lines come from?

probably from github thinking I deleted a file called "cardzone" and added a file called "card_zone" when I simply renamed and moved it like other files

ZeldaZach commented:
Can you remove the oracle/ translations as well?

ZeldaZach commented:
Can you resolve the merge conflicts?

tooomm commented on cockatrice/src/client/tabs/tab_supervisor.cpp:
```diff
@@ -11,8 +14,7 @@
 #include "pb/serverinfo_user.pb.h"
-#include "pixmapgenerator.h"
-#include "settingscache.h"
+#include "server.h"
```
Not sure about this one

tooomm reviewed (changes requested):
Great, the list of changes is now somewhat manageable. Thanks.
I had a look and put some notes when skimming trough.

Others, if you want to look at the source repo and have a look at the new structure, check if something should be changed: https://github.com/LunyaticCat/Cockatrice/tree/code-refractoring

LunyaticCat commented on cockatrice/src/dialogs/dlg_register.cpp:
```diff
@@ -351,7 +351,7 @@ DlgRegister::DlgRegister(QWidget *parent) : QDialog(parent)
     setLayout(mainLayout);
 
-    setWindowTitle(tr("Register to server"));
+    setWindowTitle(tr("DlgRegister to server"));
```
Ah yes, refactoring must have changed it without me noticing ^^'

tooomm commented:
@LunyaticCat It's now only about the conflicts that need resolution 👍 

![image](https://github.com/user-attachments/assets/8e6d10ee-fc64-40bf-9d00-19016217f3a6)

tooomm reviewed (approved):
Thanks for resolving the conflicts! 👍 

Will defer to others to sanity check the actual naming and folder structure. @ZeldaZach @ebbit1q

ZeldaZach reviewed (approved):
If you can untrack the changes on the `.github` and `webclient` dirs, this is good to go

ZeldaZach commented:
Can you resolve conflicts one more time? Then I'll merge this through!

LunyaticCat commented:
> Can you resolve conflicts one more time? Then I'll merge this through!

ok I think I did it!
